gajim: update to the current @1.3.3

barracuda156 opened the pull request:
#### Description

Update `gajim` to the current version.

###### Type(s)
<!-- update (title contains ": U(u)pdate to"), submission (new Portfile) and CVE Identifiers are auto-detected, replace [ ] with [x] to select -->

- [ ] bugfix
- [x] enhancement
- [ ] security fix

###### Tested on
<!-- Triple-click and copy the next line and paste it into your shell. It will copy your OS and Xcode version to the clipboard. Paste it here replacing this section.
sh -c 'echo "macOS $(sw_vers -productVersion) $(sw_vers -buildVersion) $(uname -m)"; xcode=$(xcodebuild -version 2>/dev/null); if [ $? == 0 ]; then echo "$(echo "$xcode" | awk '\''NR==1{x=$0}END{print x" "$NF}'\'')"; else echo "Command Line Tools $(pkgutil --pkg-info=com.apple.pkg.CLTools_Executables | awk '\''/version:/ {print $2}'\'')"; fi' | tee /dev/tty | pbcopy
-->
macOS 10.6.8 Server
Xcode 3.2.6

macOS 10.6 PPC (10A190)
Xcode 3.2

barracuda156 commented:
@kencu @dbevans I got a question re this port actually. `Readme.md` lists the following runtime deps:
```
- python3.7 or higher
- python3-gi
- python3-gi-cairo
- gir1.2-gtk-3.0 (>=3.22)
- python3-nbxmpp (>=2.0.4)
- python3-openssl (>=16.2)
- python3-css-parser
- python3-keyring
- python3-precis-i18n
- python3-packaging
- python3-setuptools
- gir1.2-soup-2.4
- GLib (>=2.60.0)
```
Now, we do not have anything looking like `python3-precis-i18n`, and I am not 100% sure re `python3-css-parser`.
`python3-gi` is apparently `py-gobject3`.

reneeotten reviewed (changes requested):
@barracuda156 you're not really making use of the "python" PG except for adding it. 

You'd probably look into setting "python.default_version" early on and then not hardcoding the version number in the dependencies. Also things like ${python.bin} and variables for frameworks are defined in the PG and there is no need to do these things again manually in the Portfile. 
Finally, if possible switch to the default Python version (i.e., 3.10) it the package works with it.

reneeotten commented on net/gajim/Portfile:
```diff
@@ -16,37 +17,73 @@ long_description    The goal of Gajim is to provide a full featured and \
 maintainers         nomaintainer
 
-platforms           darwin
-supported_archs     noarch
```
is this not "noarch" anymore, does the package compile extensions now?

reneeotten commented on net/gajim/Portfile:
```diff
@@ -1,10 +1,11 @@
-set branch          [join [lrange [split ${version} .] 0 1] .]
+version             1.3.3-2
+revision            0
 homepage            http://www.gajim.org/
```
please use https

reneeotten commented on net/gajim/Portfile:
```diff
@@ -16,37 +17,73 @@ long_description    The goal of Gajim is to provide a full featured and \
+variant zeroconf {
+    depends_run port:avahi \
+                port:py27-pybonjour
+    # there is no py38--pybonjour atm
```
this should likely not be the PY27 version?

barracuda156 replied:
@reneeotten We only got this one: https://ports.macports.org/port/py-pybonjour/details/

reneeotten replied:
then you either remove the variant or add the missing package. This will never work since you use Python 3.8 and them add a PY27 package here...

barracuda156 replied:
> then you either remove the variant or add the missing package. This will never work since you use Python 3.8 and them add a PY27 package here...

Fair enough. I have submitted PR for `py-pybonjour`: https://github.com/macports/macports-ports/pull/14781

reneeotten commented on net/gajim/Portfile:
```diff
@@ -16,37 +17,73 @@ long_description    The goal of Gajim is to provide a full featured and \
+depends_lib         path:lib/pkgconfig/gtk+-3.0.pc:gtk3 \
+                    port:glib2 \
+                    port:libsoup \
+                    port:py38-gobject3
```
without looking in detail, but just quickly glancing over the installation instructions on https://dev.gajim.org/gajim/gajim/-/wikis/help/Gajim-on-macOS, it seems that there are some dependencies missing?

barracuda156 replied:
@reneeotten This is what I am asking above: https://github.com/macports/macports-ports/pull/14671#issuecomment-1107515347

reneeotten replied:
the list I mention has other, non-Python packages as well I think

reneeotten replied:
oh, of course the the PyPI names are listed in upstream's setup.py

barracuda156 replied:
`py-nbxmpp` dependency is horribly outdated, bumping: https://github.com/macports/macports-ports/pull/14782

barracuda156 commented:
> @barracuda156 you're not really making use of the "python" PG except for adding it.
> 
> You'd probably look into setting "python.default_version" early on and then not hardcoding the version number in the dependencies. Also things like ${python.bin} and variables for frameworks are defined in the PG and there is no need to do these things again manually in the Portfile. Finally, if possible switch to the default Python version (i.e., 3.10) it the package works with it.

@reneeotten Thank you, I will fix Python handling.

barracuda156 commented:
@reneeotten Could you help me, what the buildbot dislikes?
```
Run sudo find \
[10](https://github.com/macports/macports-ports/runs/6335881051?check_suite_focus=true#step:9:10)
find: /tmp/mpbb: No such file or directory
[11](https://github.com/macports/macports-ports/runs/6335881051?check_suite_focus=true#step:9:11)
Error: Process completed with exit code 1.
```

barracuda156 commented:
UPD: This is confusing since now Macports fails to parse `py-gst-python` portfile for me locally, even exactly identical portfile which it has already built successfully. The following is extracted from Macports software package (I added extension and removed initial +):
[PORTFILE.txt](https://github.com/macports/macports-ports/files/8645741/PORTFILE.txt)

kencu commented:
1. what error do you see?
2. you should append to the configure.pre_args and configure.args, so you don't overwrite everything already there
3. you don't need that destroot.cmd I believe - it's the default
4. you don't need use_configure yes -- it's the default

barracuda156 commented:
@kencu Now I get `Failed to parse file python/py-gst-python/Portfile: can't read "python.version": no such variable` when running port sync. I cannot figure out why though. I have already built these earlier:
```
36-111% port -v installed | grep gst-python
  py38-gst-python @1.20.2_0 (active) requested_variants='' platform='darwin 10' archs='ppc' date='2022-05-08T01:29:38+0800'
  py39-gst-python @1.20.2_0 (active) requested_variants='' platform='darwin 10' archs='ppc' date='2022-05-08T01:28:03+0800'
```

reneeotten commented on python/py-gst-python/Portfile:
```diff
@@ -21,19 +21,20 @@ checksums           rmd160  38becce2998c4d5909bc3493c124d850925de473 \
 
+if {${name} ne ${subport}} {
+depends_build       port:pkgconfig
+depends_lib         port:py${python.version}-gobject3 \
```
please always use `depends_build-append` and `depends_lib-append` so that you are not overwriting dependencies that are added by the PortGroupts

reneeotten replied:
also, please indent these lines properly

reneeotten commented on python/py-gst-python/Portfile:
```diff
@@ -21,19 +21,20 @@ checksums           rmd160  38becce2998c4d5909bc3493c124d850925de473 \
+depends_build       port:pkgconfig
+depends_lib         port:py${python.version}-gobject3 \
+                    port:gstreamer1-gst-plugins-base
 livecheck.type      none
```
this is the default set by the `python` PG so please remove the `livecheck.type` line

reneeotten reviewed (changes requested):
the `set my_name` statement is useless, please remove; the `distname` field can also be removed as it is the same as the default set by the `python` PG. If you want you can use `python.rootname` in the `master_sites` field or just spell it out.

reneeotten commented:
@barracuda156 I update to the latest upstream version, cleaned-up the Portfiles, added the new dependency. The program starts, but I other than that I have no idea what to do with it ;) 

Please give it a try and, if possible, verify that the variants are still relevant/needed. If needed, can you please provide a description for each of them as to silence the `lint` warnings?

mascguy commented:
A few comments, in no particular order...

For `py-gst-python`:
* The port needs a lib dependency on `gstreamer1`, based on a quick test with trace mode enabled. (You can drop the lib dep for the plugins base port.)
* The version you're trying to publish - 1.20.2 - requires a newer version of `gstreamer1`. (Per the log: `Dependency gstreamer-1.0 found: NO found 1.16.2 but need: '>= 1.20.0'`.) But that stack is very complicated to update, which is why it hasn't been done. In short, downgrade this port to 1.16.2, and test to be sure it works with our current `gstreamer1` port.

For `py-css-parser`
* This port is outdated; latest is 1.0.9. Once you downgrade `py-gst-python`, be sure that the latter works with the latest version of `py-css-parser`. if not, you'll need to downgrade this one too.

barracuda156 commented:
> A few comments, in no particular order...
> 
> For `py-gst-python`:
> 
> * The port needs a lib dependency on `gstreamer1`, based on a quick test with trace mode enabled. (You can drop the lib dep for the plugins base port.)
> * The version you're trying to publish - 1.20.2 - requires a newer version of `gstreamer1`. (Per the log: `Dependency gstreamer-1.0 found: NO found 1.16.2 but need: '>= 1.20.0'`.) But that stack is very complicated to update, which is why it hasn't been done. In short, downgrade this port to 1.16.2, and test to be sure it works with our current `gstreamer1` port.

@mascguy I actually have it updated locally (moved to `mesonbuild` etc.). May not be the latest versions already, I did updates a while ago, but way ahead of Macports. That is why, apparently, `gajim` update was just fine locally with all used versions of dependencies.

barracuda156 commented:
Closing since: https://github.com/macports/macports-ports/commit/b8ad888e64a7aac5d6e0de1dd126e21ea4a60fab
